tests/scene/test_viewport.h

```diff
@@ -1432,6 +1433,12 @@ int TestArea2D::counter = 0;
 TEST_CASE("[SceneTree][Viewport] Physics Picking 2D") {
 	// FIXME: MOUSE_MODE_CAPTURED if-conditions are not testable, because DisplayServerMock doesn't support it.
 
+	// NOTE: This test requires a real physics server.
+	PhysicsServer2D *physics_server_2d_dummy = dynamic_cast<PhysicsServer2DDummy *>(PhysicsServer2D::get_singleton());
```

Comment:
I believe this would be the idiomatic Godot way to check this?
```suggestion
	PhysicsServer2DDummy *physics_server_2d_dummy = Object::cast_to<PhysicsServer2DDummy>(PhysicsServer2D::get_singleton());
```